### PR TITLE
feat: Add type annotations and fix mypy errors

### PR DESCRIPTION
**Pull request by gabezurita**
Adds type annotations throughout the codebase and removes the mypy.ini file that was previously disabling type checking. Key changes include:
- Remove mypy.ini as all type checking errors are now fixed
- Add type annotations to all functions in test files
- Add return type annotations for test functions (-> None)
- Add type hints for function parameters and variables
- Create TypedDict for structured dictionary types
- Fix generic type parameters for lists and dictionaries
- Add type annotations for mock objects and fixtures

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -48,19 +49,17 @@ def common_diagnostic_codes():
...
         ),
         "contention_csv": (
-            "CONTENTION TEXT,CLASSIFICATION CODE,CLASSIFICATION TEXT\n" "PTSD,8989,Mental Disorders\n" "Knee pain,8997,Knee\n"
+            "CONTENTION TEXT,CLASSIFICATION CODE,CLASSIFICATION TEXT\nPTSD,8989,Mental Disorders\nKnee pain,8997,Knee\n"
         ),
         "logging_csv": (
-            "Autosuggestion Name,Other Columns\n"
-            "Tinnitus (ringing in ears),data\n"
-            "PTSD (post-traumatic stress disorder),data\n"
+            "Autosuggestion Name,Other Columns\nTinnitus (ringing in ears),data\nPTSD (post-traumatic stress disorder),data\n"
```
Comment (brostk) [inline tests/conftest.py:62]: Nit: I find the previous formatting more readable, curious what other people think.

File: tests/test_brd_classification_codes.py [hunk 2]
```diff
@@ -8,9 +10,16 @@
 from tests.conftest import json, patch, pytest
 
 
-def test_get_classification_names_by_code(common_classification_codes, mock_file_open):
+class ItemDict(TypedDict):
+    id: int
+    name: str
```
Comment (brostk) [inline tests/test_brd_classification_codes.py:15]: Is there an advantage between a new class here vs. using Dict[int, str]?

File: tests/test_logging.py [hunk 3]
```diff
@@ -124,21 +125,21 @@ def test_log_contention_stats_expanded(mocked_func):
...
-        "contention_text": "unmapped contention text ['acl tear']",
+        "contention_text": "acl tear right",
         "diagnostic_code": "None",
         "is_in_dropdown": False,
         "is_lookup_table_match": True,
         "is_multi_contention": False,
         "endpoint": "/expanded-contention-classification",
-        "processed_contention_text": "acl tear",
         "classification_method": "contention text",
+        "processed_contention_text": "acl tear",
```
Comment (brostk) [inline tests/test_logging.py:135]: comma at the end of the list; also curious about the mapped value being changed on line 128.

File: tests/test_logging.py [hunk 4]
```diff
@@ -167,20 +168,20 @@ def test_non_classified_contentions(mocked_func):
...
-        "contention_text": "unmapped contention text",
+        "contention_text": "John has an acl tear right",
         "diagnostic_code": "None",
         "is_in_dropdown": False,
         "is_lookup_table_match": False,
         "is_multi_contention": False,
         "endpoint": "/expanded-contention-classification",
+        "classification_method": "not classified",
         "processed_contention_text": None,
-        "classification_method": classified_by,
```
Comment (brostk) [inline tests/test_logging.py:177]: Same as above - changed values

File: tests/test_logging.py [hunk 5]
```diff
@@ -223,22 +224,22 @@ def test_multiple_contentions(mocked_func):
...
-            "contention_text": "unmapped contention text ['anxiety']",
+            "contention_text": "anxiety condition due to somethiing that happened",
             "diagnostic_code": "None",
             "is_in_dropdown": False,
             "is_lookup_table_match": True,
             "is_multi_contention": True,
             "endpoint": "/expanded-contention-classification",
+            "classification_method": "contention_text",
             "processed_contention_text": "anxiety",
-            "classification_method": classified_by,
```
Comment (brostk) [inline tests/test_logging.py:241]: Same as above; typo on line 235 "somethiing"

Comment (gabezurita) [reply tests/test_logging.py:241]: Interesting! That typo is on `main`!
https://github.com/department-of-veterans-affairs/contention-classification-api/blob/50a402239541ca2d03155ae4f83dd1cedafc2ab2/tests/test_logging.py#L193

We can fix here

**Comment by gabezurita** [comment]
Note: the below refactor PR will mean we re-do this one!
https://github.com/department-of-veterans-affairs/contention-classification-api/pull/19#pullrequestreview-2548121521

**Comment by gabezurita** [comment]
Closing in favor of post-refactor PR:
https://github.com/department-of-veterans-affairs/contention-classification-api/pull/24/files
